refactor(core): migrations should parse cli workspace config as json5

devversion opened the pull request:
Currently we try to parse CLI workspace configurations gracefully by
using the native `JSON.parse()` method. This means that the CLI workspace
configuration needs to follow the strict JSON specification because otherwise
the migrations would not be able to find TypeScript configurations in the CLI
project where JSON5 workspace configurations are supported.

In order to handle such workspace configurations, we leverage the JSON
parsing logic from `@angular-devkit/core` which is also used by the CLI.

alfaproject commented on packages/core/schematics/test/project_tsconfig_paths_spec.ts:
```diff
@@ -25,6 +25,26 @@ describe('project tsconfig paths', () => {
+  it('should be able to read workspace configuration which is using JSON5 features', () => {
+    testTree.create('/my-build-config.json', '');
+    testTree.create('/angular.json', `{
+      // Comments or trailing commands are only supported in JSON5.
```
```suggestion
      // Comments or trailing commas are only supported in JSON5.
```

devversion replied:
ups. thanks! definitely meant commas :smile:

devversion committed:
refactor(core): migrations should parse cli workspace config as json5

Currently we try to parse CLI workspace configurations gracefully by
using the native `JSON.parse()` method. This means that the CLI workspace
configuration needs to follow the strict JSON specification because otherwise
the migrations would not be able to find TypeScript configurations in the CLI
project where JSON5 workspace configurations are supported.

In order to handle such workspace configurations, we leverage the JSON
parsing logicfrom the `@angular-devkit/core` which is also used by the CLI.

CaerusKaru reviewed (approved):
LGTM

IgorMinar reviewed (approved):
lgtm. it's a bit surprising that we missed this earlier. we should start using the workspace apis in the future.

IgorMinar commented:
thanks Paul!

devversion commented:
@IgorMinar There were some concerns with using the workspace utility functions which is why we created our own logic for gracefully parsing the workspace configs.ä (we used workspace utilities in Material initially as well)

Concerns were that in older CLI projects the right version of the devkit utilities / `@schematics/angular` utilities is not guaranteed and that we can't depend on these as ee also want to support projects upgrading directly from v5 to v8 (for example)
